### PR TITLE
Implementation of opcodes in LC-3 VM

### PR DESCRIPTION
**Pull request by iinaki**
## Summary
The first main problem that one is encountered when developing a VM is implementing the various opcodes that it supports. In the case of the LC-3 architecture there are 16 different opcodes, and everything the computer can calculate is some sequence of these simple instructions. My main goal thus with this PR is to implement each opcode and test they work well.

## Changes 
- Created the `operations` directory, it contains a module for each of the opcodes and their unit tests
- Added Makefile and instructions on how to use it on the readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,12 @@
...
+# VM that uses LC-3 architecture
+
+Following [this tutorial.](https://www.jmeiners.com/lc3-vm/)
+
+## OP Codes
+
+
+### How to run
+- Clone the repository with `git clone`.
+- Enter the directory containing the VM files and run `make run`.
```
Comment (JereSalo) [inline README.md]: I would clarify here that `make run` doesn't work yet. And then when implemented remove the clarification.

File: src/operations/op_add.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
+use crate::register::Register;
+
+use super::{sign_extend, update_flags};
+
+pub fn op_add(register: &mut Register, instr: u16) {
```
Comment (JereSalo) [inline src/operations/op_add.rs]:
```suggestion
pub fn op_add(registers: &mut Registers, instr: u16) {
```
I would make this change here and in every other opcode, it is more accurate to have the name in plural. Also changing the struct's name.

**Review by JereSalo — APPROVED** [review]
I like how you’ve split the code into different files. The testing is also quite exhaustive.

File: README.md (hunk 1 above)
Comment (iinaki) [reply README.md]: Done!

File: src/operations/op_add.rs (hunk 2 above)
Comment (iinaki) [reply src/operations/op_add.rs]: Done!

**Review by JereSalo — APPROVED** [review]
lgtm

**Pull request merged**